WalletWasabi/Blockchain/TransactionBuilding/BnB/PruneByBestStrategy.cs

```diff
@@ -0,0 +1,59 @@
+using System.Linq;
+
+namespace WalletWasabi.Blockchain.TransactionBuilding.BnB;
+
+/// <summary>
+/// Strategy that stores the best solution found along the way and prunes
+/// search space where we would find worse solutions that we already have.
+/// </summary>
+public class PruneByBestStrategy : ISearchStrategy
```

Comment:
`BestStrategy` is ambiguous to me. It can be the one with less input or the one with the lowest tolerance. As far as I can see this is not implementing any. I would calculate the total extra cost and minimalize that (input fees + tolerance).

Can you implement the one and only final strategy here?

Reply:
I have temporarily renamed the class. But I need to take into account inputs' spending costs so it can change then again.